test/gurka/test_reach.cc

```diff
@@ -0,0 +1,233 @@
+
+  auto costing = factory.Create(co);
+  auto edge = gurka::findEdgeByNodes(*reader, closure_map.nodes, "A", "B");
+
+  // check its reach
+  loki::Reach reach_checker;
+  auto reach = reach_checker(std::get<1>(edge), std::get<0>(edge), 50, *reader, costing);
+
+  // all edges should have the same in/outbound reach
+  EXPECT_EQ(reach.inbound, reach.outbound);
+  EXPECT_EQ(reach.inbound, 9);
+  EXPECT_EQ(reach.outbound, 9);
+}
+
+TEST_F(TestReach, ReachWithClosures) {
+  LiveTrafficCustomize close_edge = [](baldr::GraphReader& reader, baldr::TrafficTile& tile,
+                                       uint32_t index, baldr::TrafficSpeed* current) -> void {
+    close_bidir_edge(reader, tile, index, current, "AB", closure_map);
+    close_bidir_edge(reader, tile, index, current, "BC", closure_map);
+    close_bidir_edge(reader, tile, index, current, "AD", closure_map);
+    close_bidir_edge(reader, tile, index, current, "BE", closure_map);
+  };
+  test::customize_live_traffic_data(closure_map.config, close_edge);
+
+  // Create costing
+  // - type: auto
+  // - flow mask: "default" (includes current) to consider live speeds
+  // - filter_closures: enabled, to remove closed edges
+  sif::CostFactory factory;
+  loki::Reach reach_checker;
+
+  CostingOptions co;
+  co.set_costing(valhalla::auto_);
+  co.set_flow_mask(kDefaultFlowMask);
+  co.set_filter_closures(true);
+  // Check reach for AB - enclosed by closures, so reach should be 0
+  {
+    auto costing = factory.Create(co);
+    auto edge = gurka::findEdgeByNodes(*reader, closure_map.nodes, "A", "B");
+    auto reach = reach_checker(std::get<1>(edge), std::get<0>(edge), 50, *reader, costing);
+
+    EXPECT_EQ(reach.inbound, reach.outbound);
+    EXPECT_EQ(reach.inbound, 0);
+    EXPECT_EQ(reach.outbound, 0);
+  }
+  // Check reach for HI - open, but reduced reach without ignoring closures
+  {
+    auto costing = factory.Create(co);
+    auto edge = gurka::findEdgeByNodes(*reader, closure_map.nodes, "H", "I");
+    auto reach = reach_checker(std::get<1>(edge), std::get<0>(edge), 50, *reader, costing);
+
+    EXPECT_EQ(reach.inbound, 7);
+    EXPECT_EQ(reach.outbound, 7);
+  }
+
+  // Check reach for AB ignoring closures
+  {
+    co.set_filter_closures(false);
+    auto costing = factory.Create(co);
+    auto edge = gurka::findEdgeByNodes(*reader, closure_map.nodes, "A", "B");
+    auto reach = reach_checker(std::get<1>(edge), std::get<0>(edge), 50, *reader, costing);
+
+    EXPECT_EQ(reach.inbound, 8);
+    EXPECT_EQ(reach.outbound, 8);
+  }
+  // Check reach for HI ignoring closures
+  {
+    co.set_filter_closures(false);
+    auto costing = factory.Create(co);
+    auto edge = gurka::findEdgeByNodes(*reader, closure_map.nodes, "H", "I");
+    auto reach = reach_checker(std::get<1>(edge), std::get<0>(edge), 50, *reader, costing);
+
+    EXPECT_EQ(reach.inbound, 7);
+    EXPECT_EQ(reach.outbound, 7);
+  }
+}
+
+TEST_F(TestReach, DISABLED_ReachWithClosures2) {
```

Comment:
Currently disabled. This tests the scenario where we're currently under-counting the inbound reach